src/daemon/gnet/pool.go

```diff
@@ -373,7 +378,7 @@ func (pool *ConnectionPool) handleConnection(conn net.Conn, solicited bool) {
 	q := make(chan struct{}, 1)
 	qc <- q
 	<-q
-
+	wg.Wait()
```

Comment:
Can remove qc/q, instead `close(msgC)` after err read from `errC`

Reply:
`close(msgC)` might cause `panic: send on closed channel`  in readLoop

Reply:
readLoop needs to check a quit channel then, and close msgC when its done